File: backend/ŠišAppApi/Models/Notification.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace ŠišAppApi.Models;

public class Notification
{
    [Key]
    public int Id { get; set; }
    
    [Required]
    public int UserId { get; set; }
    
    [ForeignKey("UserId")]
    public User User { get; set; } = null!;
    
    [Required]
    [MaxLength(20)]
    public string Type { get; set; } = string.Empty; // Email, SMS, InApp
    
    [Required]
    [MaxLength(500)]
    public string Message { get; set; } = string.Empty;
    
    public string? Data { get; set; } // JSON s dodatnim podacima
    
    public bool IsRead { get; set; } = false;
    
    public DateTime? ReadAt { get; set; }
    
    [Required]
    public DateTime SentAt { get; set; } = DateTime.UtcNow;
    
    public DateTime? UpdatedAt { get; set; }
    
    public bool IsDeleted { get; set; } = false;
    
    public DateTime? DeletedAt { get; set; }
} 

File: sisapp_frontend/lib/models/notification.dart
import 'package:sisapp_frontend/utils/api_datetime.dart';

class Notification {
  final int id;
  final int userId;
  final String type;
  final String title;
  final String message;
  final String? data;
  final bool isRead;
  final DateTime sentAt;
  final DateTime? readAt;

  Notification({
    required this.id,
    required this.userId,
    required this.type,
    required this.title,
    required this.message,
    this.data,
    required this.isRead,
    required this.sentAt,
    this.readAt,
  });

  factory Notification.fromJson(Map<String, dynamic> json) {
    return Notification(
      id: json['id'],
      userId: json['userId'],
      type: json['type'],
      title: (json['title'] ?? json['message'] ?? 'Obavještenje').toString(),
      message: json['message'],
      data: json['data'],
      isRead: json['isRead'],
      sentAt: ApiDateTime.parse(json['sentAt']),
      readAt: ApiDateTime.parseNullable(json['readAt']),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'userId': userId,
      'type': type,
      'title': title,
      'message': message,
      'data': data,
      'isRead': isRead,
      'sentAt': ApiDateTime.toUtcIso(sentAt),
      'readAt': readAt != null ? ApiDateTime.toUtcIso(readAt!) : null,
    };
  }
}
